README.md

```diff
@@ -239,24 +239,463 @@ One can also do it manually:
+    - Coding/decoding events using [FsCodec](https://github.com/jet/fscodec). (because Events are things that **have happened**, validating them is not a central concern as we load and fold these incontrovertible Facts)
+    - Model binding in ASP.NET (because the types de-sugar to the primitives, no special support is required). _Unlike events, there are more considerations in play in this context though; often you'll want to apply validation to the inputs (representing Commands) as you map them to [Value Objects](https://martinfowler.com/bliki/ValueObject.html), [Making Illegal States Unrepresentable](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/). Often, Single Case Discriminated Unions can be a better tool inb that context_
 
-## Managing Projections and Reactions with Equinox, Propulsion and FsKafka
+### ✅ DO Have a helper `module` per id type
+
+TODO write this up
+
+### CONSIDER UMX `Guid`s for serialized ids
+
+TODO write this up
+
+### CONSIDER UMX `strings` for serialized ids
+
+TODO write this up
+
+## Code structure
+
+### 1. `module Aggregate`
 
 <a name="aggregate-module"></a>
-## Aggregate module conventions
+#### ✅ DO stick to the `module <Aggregate>` conventions
 
 There are established conventions documented in [Equinox's `module Aggregate` overview](https://github.com/jet/equinox/blob/master/DOCUMENTATION.md#aggregate-module)
 
+#### ❌ DONT split the `module <Aggregate>`
+Having the Event Contracts, State and Decision logic in a single module can feel wrong when you get over e.g. 1000 lines of code; instincts to split the file on some basis will kick in. Don't do it; splitting the file is hiding complexity under the carpet.
+
+#### ❌ DONT move the `module Events` out
+
+The Event Contracts are the most important contract that an Aggregate has - decision logic will churn endlessly. You might even implement logic against it in other languages. But the Event Contracts you define are permanent. As a developer fresh to a project, the event contracts are often the best starting point as you try to understand what a given aggregate is responsible for. 
+
+#### ❌ DONT move the `module State`, or `evolve` logic out
+
+The State type and the associated `evolve` and `fold` functions are intimately tied to the Event Contracts. Over time, ugliness and upconversion can lead to noise, and temptation to move it out. Don't do it; being able to understand the full coupling is critical to understanding how things work, and equally critical to being able to change or add functions.
+
+#### ❌ DONT move the decision logic out
```

Comment:
![image](https://github.com/jet/dotnet-templates/assets/3721521/c90f4934-461e-4331-99e3-866ad9b6ea27)

Reply:
it wasn't quite dead IMO https://github.com/jet/dotnet-templates/pull/133/commits/4ba2cae305fa7474f46c4969fdbd339379358e73